store/store.go

```diff
@@ -146,6 +146,19 @@ type Store interface {
 	// IdentityCounts returns the number of identities stored in the
 	// store split by provider ID.
 	IdentityCounts(ctx context.Context) (map[string]int, error)
+
+	// AddMFACredential stores the specified multi-factor credential.
+	AddMFACredential(ctx context.Context, cred MFACredential) error
```

Comment:
I'm in two minds about adding to the store interface in this way. The intention was that, in general, the identity providers would use the ProviderInfo field to store provider-specific information about an identity. Here however it's not so clear-cut and maybe this is sufficiently complex to warrant new store interfaces.

However, if this really does need a new store interface we should probably move the bulk of the MFA code to be part of the main application, rather than hidden in an IDP.